optimum/neuron/exporter/convert.py

```diff
@@ -0,0 +1,279 @@
+    neuron_model = torch.jit.load(neuron_model_path)
+    neuron_outputs = neuron_model(*neuron_inputs)
+
+    # Check if we have a subset of the keys into neuron_outputs against ref_outputs
+    ref_output_names_set, neuron_output_names_set = set(ref_outputs.keys()), set(neuron_named_outputs)
+    if not neuron_output_names_set.issubset(ref_output_names_set):
+        raise OutputMatchError(
+            "Neuron model output names do not match reference model output names.\n"
+            f"Reference model output names: {ref_output_names_set}\n"
+            f"Neuron model output names: {neuron_output_names_set}"
+            f"Difference: {neuron_output_names_set.difference(neuron_output_names_set)}"
+        )
+    else:
+        neuron_output_names = ", ".join(neuron_output_names_set)
+        logger.info(f"\t-[✓] Neuron model output names match reference model ({neuron_output_names})")
+
+    # Check if the number of outputs matches the number of output names
+    if len(neuron_output_names_set) != len(neuron_outputs):
+        raise OutputMatchError(
+            f"The exported Neuron model has {len(neuron_outputs)} outputs while {len(neuron_output_names_set)} are expected."
+        )
+
+    # Check the shape and values match
+    shape_failures = []
+    value_failures = []
+    for name, output in zip(neuron_output_names_set, neuron_outputs):
+        ref_output = ref_outputs[name].numpy()
+        output = output.numpy()
+
+        logger.info(f'\t- Validating Neuron Model output "{name}":')
+
+        # Shape
+        if not output.shape == ref_output.shape:
+            logger.error(f"\t\t-[x] shape {output.shape} doesn't match {ref_output.shape}")
+            shape_failures.append((name, ref_output.shape, output.shape))
+        else:
+            logger.info(f"\t\t-[✓] {output.shape} matches {ref_output.shape}")
+
+        # Values
+        if not np.allclose(ref_output, output, atol=atol):
+            max_diff = np.amax(np.abs(ref_output - output))
+            logger.error(f"\t\t-[x] values not close enough, max diff: {max_diff} (atol: {atol})")
+            value_failures.append((name, max_diff))
+        else:
+            logger.info(f"\t\t-[✓] all values close (atol: {atol})")
+
+    if shape_failures:
+        msg = "\n".join(f"- {t[0]}: got {t[1]} (reference) and {t[2]} (neuron)" for t in shape_failures)
+        raise ShapeError("Output shapes do not match between reference model and the Neuron exported model:\n" "{msg}")
+
+    if value_failures:
+        msg = "\n".join(f"- {t[0]}: max diff = {t[1]}" for t in value_failures)
+        logger.warning(
+            "The maximum absolute difference between the output of the reference model and the Neuron "
+            f"exported model is not within the set tolerance {atol}:\n{msg}"
+        )
+
+
+def export(
+    model: "PreTrainedModel",
+    config: "NeuronConfig",
+    output: Path,
+    input_shapes: Optional[Dict[str, Tuple[int, ...]]] = None,
```

Comment:
Why is this needed?
The shape should already be in the `config`.

Reply:
Yep, may bad. Will use the config.